src/nvofapi/nvofapi_common.cpp

```diff
@@ -0,0 +1,218 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: MIT
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+#include "util/util_log.h"
+#include "util/util_string.h"
+#include "util/util_statuscode.h"
+
+#include "../inc/nvofapi/nvOpticalFlowVulkan.h"
+
+#include "nvofapi_image.h"
+#include "nvofapi_instance.h"
+
+using namespace dxvk;
+namespace nvofapi {
+
+    typedef struct NV_OF_PRIV_DATA {
+        uint32_t size;
+        uint32_t id;
+        void* data;
+    } NV_OF_PRIV_DATA;
+
+    typedef struct NV_OF_EXECUTE_PRIV_DATA_INPUT_MIPS {
+        NvOFGPUBufferHandle input[6];
+        NvOFGPUBufferHandle reference[6];
+        uint8_t reserved[100];
+    } NV_OF_EXECUTE_PRIV_DATA_INPUT_MIPS;
+
+    constexpr uint32_t NV_OF_EXECUTE_PRIV_DATA_ID_INPUT_MIPS = 6;
+
+    uint32_t NvOFInstance::GetVkOFAQueue() {
+        uint32_t count = 0;
+        m_vkGetPhysicalDeviceQueueFamilyProperties(m_vkPhysicalDevice, &count, nullptr);
+        VkQueueFamilyProperties* queueFamProps = (VkQueueFamilyProperties*)calloc(sizeof(VkQueueFamilyProperties), count);
```

Comment:
I'll leave this to Jens but I believe using `std::vector` is preferable over manual `malloc`/`calloc` and `free`. It also lets us ignore the mythical allocation failures without causing segfaults.